notification/events.go

```diff
@@ -84,40 +84,42 @@ type notificationHandler struct {
 	Ring *events.EventRing
 }
 
-// Check if notification can be send in the interval based on group by, returns true if it can sent
+// Check if notification can be sent in the interval based on group by, returns true if it can be sent
 func checkRepeatInterval(ctx context.Context, n NotificationWithSpec, event models.Event) (bool, error) {
 	validKeys := map[string]string{
-		"notification_id": n.ID.String(),
-		"resource_id":     event.Properties["id"],
-		"source_event":    event.Name,
+		"resource_id":  event.Properties["id"],
+		"source_event": event.Name,
 	}
-	var clauses []clause.Expression
+
+	clauses := []clause.Expression{clause.Eq{Column: "notification_id", Value: n.ID.String()}}
```

Comment:
@moshloop I think we always add this clause since this is a per notification spec.

Reply:
If we add `count` in here this because an advanced version of our existing rate limitation.

And the existing rate limitation can instead be a global rate limit (or per channel)

Reply:
Repeat Interval is a well understood term - rate limiting serves a different purpose - If a notification doesn't have a repeat interval, or is misconfigured then it will spam channels and therefore is not something that is opt-in, it is there to protect the slack channels / email accounts from spam and endless loops